awpy/stats/win_prob.py

```diff
@@ -0,0 +1,87 @@
+"""Calculates CT & T side likeliehood of winning a round at any given point"""
+
+from typing import List, Dict, Union
+
+from awpy import Demo
+
+
+def build_feature_matrix(demo: Demo, ticks: Union[int, List[int]]) -> List[Dict[str, float]]:
+    """
+    Builds the game state feature matrix for specified ticks.
+
+    Args:
+        demo (Demo): A parsed demo object containing CS2 match data.
+        ticks (Union[int, List[int]]): A single tick or a list of ticks at which to calculate features.
+
+    Returns:
+        List[Dict[str, Union[int, str, bool]]]: A list of dictionaries where each dictionary contains the game state features at a given tick.
+    
+    Raises:
+        ValueError: If specified ticks are not found within the demo object.
+    """
+    if not demo.ticks.empty:
+        if isinstance(ticks, int):
+            ticks = [ticks]
+        
+        # Gather round numbers for each specified tick
+        round_numbers = []
```

Comment:
I think this whole block could be improved -- we could simply filter `demo.ticks` where the `.tick` is in `ticks`, then take the unique values of the round number. This way, we avoid a for loop.

```
all_ticks = demo.ticks[demo.ticks["tick"].isin(ticks)]
round_numbers = demo.ticks.round_number.unique()
```

Reply:
I have refactored the block to avoid using a for loop